### PR TITLE
PRJ: Add Procedural Macro project template

### PR DESCRIPTION
**Pull request by vlad20012**
(via cargo-generatete template https://github.com/eupn/rust-procmacro-quickstart-template)

image

### CODE REVIEW

File: src/main/kotlin/org/rust/ide/newProject/Utils.kt [hunk 1]
```diff
@@ -48,6 +48,7 @@ fun Project.makeDefaultRunConfiguration(template: RsProjectTemplate) {
     val configuration = when (template) {
         is RsGenericTemplate.CargoBinaryTemplate -> configurationFactory.createCargoRunConfiguration()
         is RsGenericTemplate.CargoLibraryTemplate -> configurationFactory.createCargoTestConfiguration()
+        is RsCustomTemplate.ProcMacroTemplate -> configurationFactory.createCargoTestConfiguration()
```
Comment (ortem) [inline src/main/kotlin/org/rust/ide/newProject/Utils.kt:51]: Note that a project generated with `ProcMacroTemplate` doesn't contain any tests so this run configuration will be useful only after a user adds `tests` module. So I'm not sure if we really should create a run configuration here

Comment (vlad20012) [reply src/main/kotlin/org/rust/ide/newProject/Utils.kt:51]: Well, ok, let's remove it for now

**Review by ortem — APPROVED** [review]

**Comment by vlad20012** [comment]
Discussed in private with @Undin that it would be more secure to fork the template repo and merge changes from the original repo manually (it is rarely updated so it shouldn't be too annoying). So I forked it to https://github.com/intellij-rust/rust-procmacro-quickstart-template and updateв the link in the PR

**Comment by vlad20012** [comment]
bors r=ortem

**Pull request merged**